spring-boot/admin/src/main/java/eu/coatrack/admin/service/OAuthUserDetailsService.java

```diff
@@ -0,0 +1,106 @@
+package eu.coatrack.admin.service;
+
+/*-
+ * #%L
+ * coatrack-admin
+ * %%
+ * Copyright (C) 2013 - 2022 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import eu.coatrack.config.github.GithubEmail;
+import lombok.AllArgsConstructor;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class OAuthUserDetailsService {
+
+    private static final String GITHUB_API_EMAIL = "https://api.github.com/user/emails";
+
+    private final OAuth2AuthorizedClientService clientService;
+    private final RestTemplate restTemplate;
+
+    private OAuth2User getLoggedInUser() {
+        return (OAuth2User) SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getPrincipal();
+    }
+
+    public String getLoginNameFromLoggedInUser() {
+        return getLoggedInUser().getAttribute("login");
+    }
+
+    public String getNameFromLoggedInUser() {
+        return getLoggedInUser().getAttribute("name");
+    }
+
+    public String getCompanyFromLoggedInUser() {
+        return getLoggedInUser().getAttribute("company");
+    }
+
+    public String getEmailFromLoggedInUser() throws JsonProcessingException {
+        String email = getLoggedInUser().getAttribute("email");
```

Comment:
Very good point @gr-hovest-atb 
I added now a comment explaining when the situation of an empty email occurs, please check it

Reply:
Thanks for the explanation, I would suggest to shorted it to make it more clear and "to the point", suggestion:

```
// The OAuth2User object will not include an email in case the user has
// set their email as private in GitHub. In this case a dedicated call
// to the GitHub API is required to get the user's email adresses
```

Reply:
The comment seems good to me, so I just replace the previous comment that I added, for your suggestion.
Based on this, this thread is resolved